fog/ledger/server/src/db_fetcher.rs

```diff
@@ -170,42 +171,32 @@ impl<
         let block_range = self.sharding_strategy.get_block_range();
         let mut next_block_index = block_range.start_block;
         loop {
-            if block_range.contains(next_block_index) {
-                loop {
-                    let num_blocks = self.load_block_data(&mut next_block_index);
+            loop {
+                let Some(num_blocks) = self.load_block_data(&mut next_block_index) else {
+                    std::thread::sleep(Self::ERROR_RETRY_FREQUENCY);
+                    continue;
+                };
```

Comment:
note this logic changed in addressing other review comments
previously the sleep was in `load_block_data()` and this would return 0
now `load_block_data()` returns an `Option<>` and leaves the logic of what to do here instead of being split between the two and `load_block_data()` relying on this function handling `0` nicely